packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,370 @@
+    AssociationManager.addLaunchNotificationConstraint(this, props);
+  }
+
+  /**
+   * Add a launch role constraint.
+   *
+   */
+  public addLaunchRole(props: SetLaunchRoleProps) {
+    InputValidator.validateLength('description', props.description, 0, 2000);
+    AssociationManager.addLaunchRoleConstraint(this, props);
+  }
+
+  /**
+   * Add a launch template constraint.
+   *
+   */
+  public addProvisioningRules(props: AddProvisioningRulesProps) {
+    InputValidator.validateLength('description', props.description, 0, 2000);
+    AssociationManager.addTemplateConstraint(this, props);
+  }
+
+  /**
+   * Add a resource template constraint.
+   *
+   */
+  public allowTagUpdates(props: AllowTagUpdatesProps) {
+    InputValidator.validateLength('description', props.description, 0, 2000);
+    AssociationManager.addResourceUpdateConstraint(this, props);
+  }
+
+  /**
+   * Add a stack set template constraint.
+   */
+  public addStackSetConstraint(props: StackSetConstraintProps) {
+    InputValidator.validateLength('description', props.description, 0, 2000);
+    AssociationManager.addStackSetConstraint(this, props);
+  }
+
+  /**
+ * add tagOptions
+ * @param tagOption
+ */
+  public addTagOptions(tagOption: TagOption) {
+    Object.keys(tagOption).forEach(key => {
+      InputValidator.validateLength('key', key, 1, 128);
+      tagOption[key].forEach(value => {
+        InputValidator.validateLength('value', value.value, 1, 256);
+      });
+    });
+    AssociationManager.associateTagOption(this, this.portfolioId, tagOption);
+  }
+
+  /**
+   * Create a unique id based off the L1 portfolio.
+   *
+   */
+  protected abstract getKeyForPortfolio(value: string): string;
+}
+
+/**
+ * Properties for a Portfolio.
+ */
+export interface PortfolioProps {
+
+  /**
+     * Enforces a particular physical portfolio name.
+     * @default <generated>
+     */
+  readonly portfolioName: string;
+
+  /**
+     * The provider name.
+     *
+     */
+  readonly providerName: string;
+
+  /**
+     * The accept language.
+     * @default
+     */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+     * Description for portfolio.
+     *
+     * @default
+     */
+  readonly description?: string;
+
+
+  /**
+     * A collection of tags attached to portfolio.
+     * @default
+     *
+     */
+  readonly tags?: Tag[];
+
+}
+
+/**
+ * A Service Catalog portfolio.
+ */
+export class Portfolio extends PortfolioBase {
+
+  /**
+     * Creates a Portfolio construct that represents an external stream.
+     *
+     * @param scope The parent creating construct (usually `this`).
+     * @param id The construct's name.
+     * @param attrs Portfolio import properties
+     */
+  public static fromPortfolioAttributes(scope: Construct, id: string, attrs: PortfolioAttributes): IPortfolio {
+    class Import extends PortfolioBase {
+      public readonly id = id
+      public readonly portfolioId = attrs.portfolioArn.split('/').pop()!
```

Comment:
Sure thing but copied from [role.ts](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-iam/lib/role.ts#L183) so that might need to be fixed too

Reply:
fixed in update

Reply:
> Sure thing but copied from [role.ts](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-iam/lib/role.ts#L183) so that might need to be fixed too

That's not the same case (the `.split('/')` is not used for ARN parsing), so no 🙂.

Reply:
got it, makes sense